framework/include/utils/MooseTypes.h

```diff
@@ -185,6 +185,12 @@ enum LineSearchType
 #endif
 };
 
+enum RegistrationType
+{
+  APPLICATION,
+  OBJECTS
+};
+
```

Comment:
I do have two functions right now: 
```
MooseApp::dynamicRegisterApps()
MooseApp::dyanmicRegisterObjects()
```
Those two functions cover apps and objects respectively. However, if we want to support all of the MOOSE systems individual or in arbitrary combinations we'll need an enum to pick which ones to register. See my comment down below.